src/NHibernate/Context/CallSessionContext.cs

```diff
@@ -1,20 +1,28 @@
 using System;
 using System.Collections;
-using System.Runtime.Remoting.Messaging;
-
+using System.Threading;
 using NHibernate.Engine;
+#if NET461
+using System.Runtime.Remoting.Messaging;
+#endif
 
 namespace NHibernate.Context
 {
+#pragma warning disable CS1574
 	/// <summary>
 	/// Provides a <see cref="ISessionFactory.GetCurrentSession()">current session</see>
 	/// for each <see cref="System.Runtime.Remoting.Messaging.CallContext"/>.
 	/// Not recommended for .NET 2.0 web applications.
 	/// </summary>
+#pragma warning restore CS1574
 	[Serializable]
 	public class CallSessionContext : MapBasedSessionContext
 	{
+		#if NET461
 		private const string SessionFactoryMapKey = "NHibernate.Context.CallSessionContext.SessionFactoryMapKey";
+		#else
+		private static readonly AsyncLocal<IDictionary> SessionFactoryMap = new AsyncLocal<IDictionary>();
+		#endif
```

Comment:
I looks strange to me to fallback to async local. Shouldn't it be considered an error to try using call context with .Net Core?

Reply:
I think it's better to provide feature-parity. The behaviour will remain consistent with Full .NET.